examples/real_data_example/src/seismic_numpy.h

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 TGS
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//    http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <absl/status/status.h>
+#include <mdio/mdio.h>
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+
+#include "tensorstore/tensorstore.h"
+
+std::string GetNumpyDtypeJson(const tensorstore::DataType& dtype) {
+  using namespace tensorstore;
+
+  // Mapping of DataTypeId to NumPy dtype JSON-like string format
+  static const std::unordered_map<DataTypeId, std::string> dtype_to_numpy_json{
+      {DataTypeId::int8_t, "|i1"},    {DataTypeId::uint8_t, "|u1"},
+      {DataTypeId::int16_t, "<i2"},   {DataTypeId::uint16_t, "<u2"},
+      {DataTypeId::int32_t, "<i4"},   {DataTypeId::uint32_t, "<u4"},
+      {DataTypeId::int64_t, "<i8"},   {DataTypeId::uint64_t, "<u8"},
+      {DataTypeId::float32_t, "<f4"}, {DataTypeId::float64_t, "<f8"},
+  };
+
+  // Find the matching NumPy dtype string or return "unknown" if not found
+  auto it = dtype_to_numpy_json.find(dtype.id());
+  if (it != dtype_to_numpy_json.end()) {
+    return it->second;
+  } else {
+    return "unknown dtype";
+  }
+}
+
+template <typename T, mdio::DimensionIndex R, mdio::ArrayOriginKind OriginKind>
+absl::Status WriteNumpy(const mdio::SharedArray<T, R, OriginKind>& accessor,
+                        const std::string& filename) {
+  auto domain = accessor.domain();
+  auto type_id = GetNumpyDtypeJson(accessor.dtype());
+
+  if (domain.rank() != 3) {
+    return absl::InvalidArgumentError("Expected a 3D array");
+  }
+
+  auto inline_inclusive_min = domain[0].inclusive_min();
+  auto inline_exclusive_max = domain[0].exclusive_max();
+
+  auto xline_inclusive_min = domain[1].inclusive_min();
+  auto xline_exclusive_max = domain[1].exclusive_max();
+
+  auto depth_inclusive_min = domain[2].inclusive_min();
+  auto depth_exclusive_max = domain[2].exclusive_max();
+
```

Comment:
```suggestion
  const mdio::Index width = inline_exclusive_max - inline_inclusive_min;
  const mdio::Index height = xline_exclusive_max - xline_inclusive_min;
  const mdio::Index depth = depth_exclusive_max - depth_inclusive_min;
```
More technically correct and explicit to use the mdio::Index type (maps back to int64_t)

Reply:
done